docroot/profiles/custom/cgov_site/modules/custom/cgov_yaml_content/src/Service/YamlEntityEmbedProcessor.php

```diff
@@ -0,0 +1,209 @@
+        if ($uuid) {
+          $drupalEntityEl->setAttr('data-entity-uuid', $uuid);
+        }
+        else {
+          $this->throwParamError("Could not find entity for embed", $entityType, $queryParams);
+        }
+      }
+    }
+
+    // Return the value.
+    $body = $doc->find('body');
+    if ($body->count() > 0) {
+      $value = $body->html();
+    }
+
+    return $value;
+  }
+
+  /**
+   * Finds a UUID for embedding.
+   *
+   * YAML Content module does not handle media in its helper, so
+   * we will need to roll our own.
+   *
+   * @param string $entity_type
+   *   The type of entity being imported.
+   * @param array $content_data
+   *   The import content structure representing the entity being searched for.
+   *
+   * @return string|false
+   *   Return a matching UUID if one is found, or FALSE otherwise.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   */
+  public function queryEntityUuid($entity_type, array $content_data) {
+    $storage = $this->entityTypeManager->getStorage($entity_type);
+    $query = $storage->getQuery('AND');
+    foreach ($content_data as $key => $value) {
+      if ($key != 'entity' && !is_array($value)) {
+        $query->condition($key, $value);
+      }
+    }
+    $entity_ids = $query->execute();
+
+    // Nothing was found.
+    if (empty($entity_ids)) {
+      return FALSE;
+    }
+
+    // If there were more than one then we should throw an error.
+    if (count($entity_ids) != 1) {
+      return $this->throwParamError('Multiple embed entities matched query.', $entity_type, $content_data);
+    }
+
+    $entities = $storage->loadMultiple($entity_ids);
+
+    if (count($entity_ids) != 1) {
+      return $this->throwParamError('Multiple embed entities matched query.', $entity_type, $content_data);
+    }
+
+    return array_shift($entities)->uuid();
+  }
+
+  /**
+   * Converts a collection of attributes to entity query params.
+   *
+   * @param \DOMNamedNodeMap $attrs
+   *   Attributes List.
+   *
+   * @return array
+   *   An array of the entity query parameters.
+   */
+  public function mapYamlQueryAttrToParams(\DOMNamedNodeMap $attrs) {
+    $params = [];
+    foreach ($attrs as $name => $attrNode) {
+      $matches = [];
+      if (preg_match("/^data-cgov-yaml-query-(.*)$/", $name, $matches)) {
+        $params[$matches[1]] = $attrNode->value;
+      }
+    }
+    return $params;
+  }
+
+  /**
+   * Prepare an error message and throw error.
+   *
+   * @param string $error_message
+   *   The error message to display.
+   * @param string $entity_type
+   *   The entity type.
+   * @param array $filter_params
+   *   The filters for the query conditions.
+   */
+  protected function throwParamError($error_message, $entity_type = "ENTITY TYPE UNKNOWN", array $filter_params = []) {
+    // Build parameter output description for error message.
+    $error_params = [
+      '[',
+      '  "entity_type" => ' . $entity_type . ',',
+    ];
```

Comment:
The value for the `$entity_type` variable will not be enclosed in quotation marks.  E.g. if `$entity_type` contains the value "foo", this will result in the second array value being `"entity_type" => foo,`. 

I believe the intent is for that value to be `"entity_type" => "foo",` (or, more likely, "node" or "media" or some real entity type).

Reply:
It is fixed. I also fixed where I copied it from.